rskj-core/src/test/java/co/rsk/rpc/modules/debug/DebugModuleImplTest.java

```diff
@@ -214,4 +216,44 @@ public void debug_traceTransaction_retrieveSimpleAccountTransferWithTraceOptions
 
         Assert.assertEquals(resultWithNoOptions, resultWithNonEmptyOptions);
     }
+
+    @Test
+    public void debug_traceBlock_retrieveUnknownBlockAsNull() throws Exception {
+        byte[] hash = stringHexToByteArray("0x00");
+        when(blockStore.getBlockByHash(hash)).thenReturn(null);
+
+        JsonNode result = debugModule.traceBlock("0x00", null);
+
+        Assert.assertNull(result);
+    }
+
+    @Test
+    public void debug_traceBlock_retrieveSimpleContractsCreationTrace() throws Exception {
+        DslParser parser = DslParser.fromResource("dsl/contracts10.txt");
```

Comment:
Newbie question - what does the transaction do exactly? Is it just a dummy compiled transaction?

Reply:
yes, correct. This produces a list of dummy transactions. We use our own DSL to describe a script in a file that defines transactions and blocks just to simplify some test cases. In this particular case I borrowed tests from the similar `trace transaction` use case from [here](https://github.com/rsksmart/rskj/blob/47c7d93315f5fa7d5eb06a4a3cab794a13a5a770/rskj-core/src/test/java/co/rsk/rpc/modules/debug/DebugModuleImplTest.java#L96).